statsd/statsd.go

```diff
@@ -552,6 +563,17 @@ func (c *Client) Gauge(name string, value float64, tags []string, rate float64)
 	return c.send(metric{metricType: gauge, name: name, fvalue: value, tags: tags, rate: rate, globalTags: c.tags, namespace: c.namespace})
 }
 
+// Gauge measures the value of a metric at a given time.
+// Even with client side aggregation enabled, there is no aggregation done on a metric
+// sent using GaugeWithTimestamp: it is written as is in the serialization buffer.
+func (c *Client) GaugeWithTimestamp(name string, value float64, tags []string, rate float64, timestamp time.Time) error {
```

Comment:
Should we take a `time.Time` or a `int` directly here ?

Reply:
I think we should go with `time.Time` just because it a public-facing API and `time.Time` is here for that, showing you _have_ to provide something which is representing time, not just an int. That's subtle but I think that's the way to go. WDYT?